File: services/gateway/gateway-api/src/main/java/com/example/gateway/api/exception/ExceptionModel.java
package com.example.gateway.api.exception;

import java.time.LocalDateTime;

public class ExceptionModel {
    private LocalDateTime timestamp;
    private ExceptionCode code;
    private String message;
    private String details;

    public ExceptionModel(ExceptionCode code) {
        this.code = code;
        this.message = code.getMessage();
        this.timestamp = LocalDateTime.now();
    }

    public ExceptionModel(ExceptionCode code, String message) {
        this.code = code;
        this.message = message;
        this.timestamp = LocalDateTime.now();
    }

    public ExceptionModel(ExceptionCode code, String message, String details) {
        this.code = code;
        this.message = message;
        this.timestamp = LocalDateTime.now();
        this.details = details;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public ExceptionCode getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public String getDetails() {
        return details;
    }
}

File: services/tradelog/tradelog-api-spec/src/main/kotlin/com/example/tradelog/api/spec/model/ExceptionModel.kt
package com.example.tradelog.api.spec.model

import com.example.tradelog.api.spec.exception.ExceptionCode
import java.time.LocalDateTime

class ExceptionModel(_code: ExceptionCode, _message: String, _details: String) {
    private val code: ExceptionCode = _code
    private val message: String = _message
    private val details: String = _details
    private val timestamp: LocalDateTime = LocalDateTime.now()
}